content/intro-to-storybook/react/ja/using-addons.md

```diff
@@ -0,0 +1,163 @@
+commit: 'b3bca4a'
+---
+
+Storybook はチームメンバーの開発を向上する堅牢な[アドオン](https://storybook.js.org/addons/introduction/)のシステムを誇っています。ここまでチュートリアルを進めてきたのであればいくつかのアドオンをすでに見ており、[テストの章](/react/ja/test/)では実際に使用していることでしょう。
+
+<div class="aside">
+<strong>アドオンの一覧を見たいですか？</strong>
+<br/>
+😍 公式にサポートされているアドオンや、コミュニティに強くサポートされているアドオンの一覧は<a href="https://storybook.js.org/addons/addon-gallery/">ここ</a>から見ることができます。
+</div>
+
+全てのアドオンの使用方法や設定方法、特定のユースケースを書きつくすことは出来ません。なので、ここでは Storybook のエコシステムの中で最も人気のアドオンである [Knobs](https://github.com/storybooks/storybook/tree/master/addons/knobs) を例にとって見てみましょう。
+
+## Knobs をセットアップする
+
+Knobs は、デザイナーや開発者がコードを書かずにコンポーネントで実験したり、触ったりできるようにするアドオンです。ストーリーに描画されるコンポーネントに対し、ユーザーが操作し、動的に定義可能なプロパティを渡せます。これが今から実装しようとするものです:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook/addon-knobs-demo.mp4"
+    type="video/mp4"
+  />
+</video>
+
+### 導入方法
+
+まずは、依存関係を追加します。
+
+```bash
+yarn add -D @storybook/addon-knobs
+```
+
+`.storybook/main.js` ファイルに Knobs を登録します。
+
+```javascript
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../src/components/**/*.stories.js'],
+  addons: [
+    '@storybook/preset-create-react-app',
+    '@storybook/addon-actions',
+    '@storybook/addon-knobs',
+    '@storybook/addon-links',
+  ],
+};
+```
+
+<div class="aside">
+<strong>📝 アドオンの記述順序には意味があります！</strong>
+<br/>
+アドオンを記載した順序はアドオンのパネル (パネルが表示されるアドオンの場合) のタブ順になります。
+</div>
+
+以上です。ストーリーで使ってみましょう。
+
+### 使い方
+
+`Task` コンポーネントに Knobs のオブジェクトを使ってみましょう。
+
+まず、Knobs の `withKnobs` デコレーターと `object` を `Task.stories.js` でインポートします。
+
+```javascript
+// src/components/Task.stories.js
+
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { withKnobs, object } from '@storybook/addon-knobs/react';
+```
+
+<div class="aside">
+もし、TypeScript を使用しているのであれば、すこし調整が必要です。
+上記のインポートの代わりに、<code>import { withKnobs, object } from '@storybook/addon-knobs'</code> と書いてください。
+</div>
+
+次に、`Task.stories.js` ファイルのデフォルトエクスポートの `decorators` キーに `withKnobs` を追加してください:
+
+```javascript
+// src/components/Task.stories.js
+
+export default {
+  component: Task,
+  title: 'Task',
+  decorators: [withKnobs],
+  excludeStories: /.*Data$/,
+};
+```
+
+最後に、Default ストーリーに Knobs の `object` を導入します:
+
+```javascript
+// src/components/Task.stories.js
+
+export const Default = () => {
+  return <Task task={object('task', { ...taskData })} {...actionsData} />;
+};
+```
+
+すると、「Knobs」という新しいタブが画面下部の「Actions」タブの隣に表示されます。
+
+[ここ](https://github.com/storybooks/storybook/tree/master/addons/knobs#object)に書かれているように、`object` はラベルとデフォルトのオブジェクトを引数として受け取ります。ラベルはアドオンパネルのテキストエリアの隣に表示されます。渡したオブジェクトはテキストエリアに編集可能な JSON として表示されます。JSON として正しい値を記載している限り、コンポーネントはそのデータに合わせて表示されます。
+
+## Storybook の範囲をアドオンが進化させる
+
+これで、Storybook が優れた [CDD の開発環境](https://blog.hichroma.com/component-driven-development-ce1109d56c8e)を提供するだけでなく、対話的なドキュメントとしても使えるようになりました。PropType も素晴らしいのですが、Knobs が実装された Storybook はデザイナーや、コンポーネントのコードを知らない人にとってコンポーネントの振る舞いを簡単に知る手段となります。
+
```

Comment:
@chameleonhead when you're able, can you adjust the CDD link to https://www.componentdriven.org/